packages/runtime/lib/runtime.js

```diff
@@ -1809,6 +1822,72 @@ class Runtime extends EventEmitter {
       await immediate()
     }
   }
+
+  async getServiceResourcesInfo (id) {
+    const workers = this.#workers.getCount(id)
+    return { workers }
+  }
+
+  async updateWorkerCount (serviceId, workers) {
+    this.#configManager.current.services.find(s => s.id === serviceId).workers = workers
+    const service = await this.#getServiceById(serviceId)
+    this.#workers.setCount(serviceId, workers)
+    service[kConfig].workers = workers
+  }
+
+  async updateServicesResources (updates) {
+    if (this.#status === 'stopping' || this.#status === 'closed') return
+
+    if (!Array.isArray(updates)) {
+      throw new errors.InvalidArgumentError('updates', 'must be an array')
+    }
+    if (updates.length === 0) {
+      throw new errors.InvalidArgumentError('updates', 'must have at least one element')
+    }
+
+    const config = this.#configManager.current
+
+    for (const update of updates) {
+      const { service: serviceId, workers } = update
+
+      if (typeof workers !== 'number') {
+        throw new errors.InvalidArgumentError('workers', 'must be a number')
+      }
+      if (!serviceId) {
+        throw new errors.InvalidArgumentError('service', 'must be a string')
+      }
+      if (workers <= 0) {
+        throw new errors.InvalidArgumentError('workers', 'must be greater than 0')
+      }
+      if (workers > MAX_WORKERS) {
+        throw new errors.InvalidArgumentError('workers', `must be less than ${MAX_WORKERS}`)
+      }
+      const serviceConfig = config.services.find(s => s.id === serviceId)
+      if (!serviceConfig) {
+        throw new errors.ServiceNotFoundError(serviceId, Array.from(this.#servicesIds).join(', '))
+      }
+
+      const { workers: currentWorkers } = await this.getServiceResourcesInfo(serviceId)
+      if (currentWorkers === workers) {
+        this.logger.warn({ serviceId, workers }, 'No change in the number of workers for service')
+        continue
+      }
+
+      if (currentWorkers < workers) {
+        await this.updateWorkerCount(serviceId, workers)
```

Comment:
What about existing workers? Are you updating the workers count?

For instance the value is used here:

https://github.com/platformatic/platformatic/blob/main/packages/runtime/lib/worker/main.js#L254

Can you please check it doesn't require update?